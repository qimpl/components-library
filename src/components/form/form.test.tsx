import React from 'react';

import { render } from '@testing-library/react';
import { withFormik } from 'formik';

import Form, { FormProps } from 'components/form';
import Input from 'components/input';

type TestFormValues = {
  firstName: string;
  lastName: string;
};

const TestForm = withFormik<FormProps, TestFormValues>({
  mapPropsToValues: () => ({
    firstName: '',
    lastName: '',
  }),
  handleSubmit: (values: TestFormValues) => JSON.stringify(values, null, 2),
})(Form);

const fields = [
  {
    id: 'firstName',
    element: <Input id='firstName' label='First name' />,
  },
  {
    id: 'lastName',
    element: <Input id='lastName' label='Last name' />,
  },
];

test('Default select', () => {
  expect(render(<TestForm fields={fields} buttonLabel='Confirm' />)).toMatchSnapshot();
});
