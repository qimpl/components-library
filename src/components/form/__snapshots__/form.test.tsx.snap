// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default select 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
      >
        <div>
          <div
            class="sc-gsTCUz eGyIgM"
          >
            <label
              for="firstName"
            >
              First name
            </label>
            <input
              id="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div>
          <div
            class="sc-gsTCUz eGyIgM"
          >
            <label
              for="lastName"
            >
              Last name
            </label>
            <input
              id="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="sc-bdfBwQ hMKfoD"
          color="#013c63"
          type="submit"
        >
          Confirm
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
    >
      <div>
        <div
          class="sc-gsTCUz eGyIgM"
        >
          <label
            for="firstName"
          >
            First name
          </label>
          <input
            id="firstName"
            type="text"
            value=""
          />
        </div>
      </div>
      <div>
        <div
          class="sc-gsTCUz eGyIgM"
        >
          <label
            for="lastName"
          >
            Last name
          </label>
          <input
            id="lastName"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="sc-bdfBwQ hMKfoD"
        color="#013c63"
        type="submit"
      >
        Confirm
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
